plugins/bufferlines/bufferline.nix

```diff
@@ -125,196 +196,465 @@ in
+      show_tab_indicators = defaultNullOpts.mkBool true ''
+        Whether to show the tab indicators.
+      '';
 
-      showDuplicatePrefix = helpers.defaultNullOpts.mkBool true "Whether to show the prefix of duplicated files.";
+      show_duplicate_prefix = defaultNullOpts.mkBool true ''
+        Whether to show the prefix of duplicated files.
+      '';
 
-      enforceRegularTabs = helpers.defaultNullOpts.mkBool false "Whether to enforce regular tabs.";
+      duplicates_across_groups = defaultNullOpts.mkBool true ''
+        Whether to consider duplicate paths in different groups as duplicates.
+      '';
 
-      alwaysShowBufferline = helpers.defaultNullOpts.mkBool true "Whether to always show the bufferline.";
+      enforce_regular_tabs = defaultNullOpts.mkBool false ''
+        Whether to enforce regular tabs.
+      '';
 
-      persistBufferSort = helpers.defaultNullOpts.mkBool true "Whether to make the buffer sort persistent.";
+      always_show_bufferline = defaultNullOpts.mkBool true ''
+        Whether to always show the bufferline.
+      '';
 
-      maxPrefixLength = helpers.defaultNullOpts.mkInt 15 "Maximum prefix length";
+      auto_toggle_bufferline = defaultNullOpts.mkBool true ''
+        Whether to automatically toggle bufferline.
+      '';
 
-      sortBy = helpers.defaultNullOpts.mkStr "id" "sort by";
+      persist_buffer_sort = defaultNullOpts.mkBool true ''
+        Whether to make the buffer sort persistent.
+      '';
 
-      diagnostics = helpers.defaultNullOpts.mkNullable (
-        with types;
-        either bool (enum [
-          "nvim_lsp"
-          "coc"
-        ])
-      ) false "diagnostics";
+      move_wraps_at_ends = defaultNullOpts.mkBool true ''
+        Whether or not the move command "wraps" at the first or last position.
+      '';
 
-      diagnosticsIndicator = helpers.defaultNullOpts.mkLuaFn null "Either `null` or a function that returns the diagnostics indicator.";
+      max_prefix_length = defaultNullOpts.mkInt 15 ''
+        Maximum prefix length used when a buffer is de-duplicated.
+      '';
 
-      diagnosticsUpdateInInsert = helpers.defaultNullOpts.mkBool true "Whether diagnostics should update in insert mode";
+      sort_by =
+        defaultNullOpts.mkNullableWithRaw
+          (
+            with types;
+            oneOf [
+              (enum [
+                "insert_after_current"
+                "insert_at_end"
+                "id"
+                "extension"
+                "relative_directory"
+                "directory"
+                "tabs"
+              ])
+              rawLua
+            ]
+          )
+          "id"
+          ''
+            How to sort the buffers.
+
+            Possible values: `insert_after_current`, `insert_at_end`, `id`, `extension`, `relative_directory`, `directory`, `tabs`.
+            Also accepts a function with a signature `function(buffer_a, buffer_b)` allowing you to compare with custom logic.
```

Comment:
Seems redundant to have the possible values in the enum type _and_ in the description.

Reply:
Maybe we could also define an actual example with a `__raw` function?

Reply:
Just to clarify, I was talking about an option example, not a settings-example. I'm fine with the way you've done it, but just as a reminder you can also do:

```nix
mkFoo' {
  description = "";
  example = {
    foo = "bar";
  };
}
```